codec-http/src/main/java/io/netty/handler/codec/http/CookieDecoder.java

```diff
@@ -290,7 +327,46 @@ private static void extractKeyValuePairs(
         }
     }
 
-    private CookieDecoder() {
-        // Unused
+    private CookieDecoder(boolean strict) {
+        this.strict = strict;
+    }
+
+    private DefaultCookie initCookie(String name, String value) {
+
+        if (name == null || name.length() == 0) {
+            logger.debug("Skipping cookie with null name");
+            return null;
+        }
+
+        if (value == null) {
+            logger.debug("Skipping cookie with null value");
+            return null;
+        }
+
+        CharSequence unwrappedValue = unwrapValue(value);
+        if (unwrappedValue == null) {
+            logger.debug("Skipping cookie because starting quotes are not properly balanced in '{}'",
+                    unwrappedValue);
+            return null;
+        }
+
+        int invalidOctetPos;
+        if (strict && (invalidOctetPos = firstInvalidCookieNameOctet(name)) >= 0) {
+            logger.debug("Skipping cookie because name '{}' contains invalid char '{}'",
+                    name, name.charAt(invalidOctetPos));
+            return null;
+        }
+
+        final boolean wrap = unwrappedValue.length() != value.length();
+
+        if (strict && (invalidOctetPos = firstInvalidCookieValueOctet(unwrappedValue)) >= 0) {
+            logger.debug("Skipping cookie because value '{}' contains invalid char '{}'",
+                    unwrappedValue, unwrappedValue.charAt(invalidOctetPos));
```

Comment:
shouldn't we better check if Logger.isDebugEnabled() for our log statements as otherwise we may do name.charAt(...) and not need it.